homeassistant/components/light/plum_lightpad.py

```diff
@@ -0,0 +1,191 @@
+"""
+Support for Plum Lightpad switches.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/light.plum_lightpad
+"""
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_HS_COLOR, SUPPORT_BRIGHTNESS, SUPPORT_COLOR, Light)
+from homeassistant.components.plum_lightpad import PLUM_DATA
+import homeassistant.util.color as color_util
+
+DEPENDENCIES = ['plum_lightpad']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Initialize the Plum Lightpad Light and GlowRing."""
+    if discovery_info is None:
+        return
+
+    plum = hass.data[PLUM_DATA]
+
+    if 'lpid' in discovery_info:
+        lightpad = plum.get_lightpad(discovery_info['lpid'])
+        async_add_entities([
```

Comment:
Add all entities in one call to `async_add_entities`. So first collect the entities in a list in the two `if` blocks and then if there are entities in the list call `async_add_entities`.

Reply:
No don't change anything but the code within `async_setup_platform`, ie make sure we only call `async_add_entities` once for each call to `async_setup_platform`.